api/v1alpha1/history_limits_test.go

```diff
@@ -0,0 +1,71 @@
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/vshn/k8up/api/v1alpha1"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHistoryLimits(t *testing.T) {
+
+	type limiter interface {
+		GetSuccessfulJobsHistoryLimit() *int
+		GetFailedJobsHistoryLimit() *int
+	}
+
+	testCases := map[string]struct {
```

Comment:
Could be simplified to `map[string]func`.

Also, I prefer to put test tables outside of test functions. I makes it more readable (subjective to me) and allows to reuse one test table in multiple test functions (yes that is actually a thing).